lib/ch/query.ex

```diff
@@ -237,7 +237,9 @@ defimpl DBConnection.Query, for: Ch.Query do
   defp encode_map_params([] = empty), do: empty
 
   defp encode_array_param(i) when is_integer(i), do: encode_param(i)
+  defp encode_array_param(f) when is_float(f), do: encode_param(f)
   defp encode_array_param(xs) when is_list(xs), do: encode_param(xs)
+  defp encode_array_param(xs) when is_tuple(xs), do: encode_param(xs)
 
   defp encode_array_param(s) do
```

Comment:
let's do the opposite:

- escape string in `encode_array_param(s)` when `is_binary(s)`
- otherwise call `encode_param`

Reply:
isn't that what was being done before?

Reply:
seems so

Reply:
but that is an issue because then stuff like `Date` does not get properly encoded

Reply:
@Zarathustra2 should map be added to special clauses as well?